beacon_node/network/src/message_handler.rs

```diff
@@ -151,58 +152,137 @@ impl<T: BeaconChainTypes + 'static> MessageHandler<T> {
 
     /// An RPC response has been received from the network.
     // we match on id and ignore responses past the timeout.
-    fn handle_rpc_response(&mut self, peer_id: PeerId, id: RequestId, response: RPCResponse) {
-        // if response id is not related to a request, ignore (likely RPC timeout)
+    fn handle_rpc_response(
+        &mut self,
+        peer_id: PeerId,
+        id: RequestId,
+        error_response: RPCErrorResponse,
+    ) {
+        //TODO: Potentially do not need to keep track of this at all. This has all been shifted
```

Comment:
Is this correct. Not using ID's anywhere else? Perhaps we should tho?

Reply:
I can't think of any scenario where sync _must_ know a request ID for a response.

Maybe they could be used for optimization, but I don't see it presently.